modules/packages/LinearAlgebra.chpl

```diff
@@ -1450,6 +1450,42 @@ private proc _isDiag(A: [?D] ?eltType) {
   return true;
 }
 
+/* Return `true` if matrix is the additive identity (zero matrix). */
+proc isZero(A: [?D] ?eltType) where isDenseMatrix(A) {
+  return _isZero(A);
+}
+
+private proc _isZero(A: [?D] ?eltType) {
```

Comment:
Why do you have a separate implementation and public proc?

Reply:
I was following the implicit convention that I saw. Such as 
```chapel
/* Return `true` if matrix is diagonal. */
proc isDiag(A: [?D] ?eltType) where isDenseMatrix(A) {
  return _isDiag(A);
}

private proc _isDiag(A: [?D] ?eltType) {
  if D.rank != 2 then
    compilerError("Rank is not 2");

  // Check if any element not along the diagonal is nonzero
  for (i, j) in D {
    if i != j && A[i, j] != 0 then return false;
  }
  return true;
}
```
where `_isDiag` is used in two different functions, one for dense matrices and one for sparse matrices.

Reply:
Okay, that makes sense. I suspect other procs do it this way so that sparse and dense versions can use the same underlying implementation. I'd keep it this way.